### ISSUE TITLE
types naming, false positive, nested BEGIN

### ISSUE THREAD

**Issue by larshp** [abaplint/abaplint#685]
`boo` is reported as wrong naming in types_naming rule, this is a false positive

```abap
    TYPES:
      BEGIN OF ty_foo,
        lazy     TYPE abap_bool,
        BEGIN OF boo,
          controller TYPE REF TO object,
        END OF boo,
      END OF ty_foo.
```
